src/System.Memory/src/System/Buffers/Text/Utf8Formatter/Utf8Formatter.Integer.Signed.Default.cs

```diff
@@ -2,47 +2,166 @@
-                if (digitCount >= buffer.Length) goto FalseExit;
+                // WriteDigits does not do bounds checks
+                if (digitCount >= buffer.Length)
+                {
+                    bytesWritten = 0;
+                    return false;
+                }
+                buffer[0] = Utf8Constants.Minus;
+                FormattingHelpers.WriteDigits((ulong)value, buffer.Slice(1, digitCount));
                 bytesWritten = digitCount + 1;
+                return true;
+            }
+            else
+            {
+                return TryFormatUInt64MultipleDigits((ulong)value, buffer, out bytesWritten);
+            }
+        }
+
+        // Split long into two parts that can each fit in a uint - {1-9 digits}{9 digits}
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool TryFormatInt64LessThanQuintillion(long value, Span<byte> buffer, out int bytesWritten)
+        {
+            if (value < 0)
+            {
+                value = -value;
+                ulong OverNineDigits = (ulong)value / Utf8Constants.Billion;
+                ulong LastNineDigits = (ulong)value - (OverNineDigits * Utf8Constants.Billion);
+
+                int digitCountOverNineDigits = FormattingHelpers.CountDigits((uint)OverNineDigits);
+                Debug.Assert(digitCountOverNineDigits >= 1 && digitCountOverNineDigits <= 9);
+                int digitCount = digitCountOverNineDigits + 9;
+                // WriteDigits does not do bounds checks
+                if (digitCount >= buffer.Length)
+                {
+                    bytesWritten = 0;
+                    return false;
+                }
                 buffer[0] = Utf8Constants.Minus;
-                buffer = buffer.Slice(1, digitCount);
+                FormattingHelpers.WriteDigits((uint)OverNineDigits, buffer.Slice(1, digitCountOverNineDigits));
+                FormattingHelpers.WriteDigits((uint)LastNineDigits, buffer.Slice(digitCountOverNineDigits + 1, 9));
+                bytesWritten = digitCount + 1;
+                return true;
             }
             else
             {
-                int digitCount = FormattingHelpers.CountDigits((ulong)value);
-                if (digitCount > buffer.Length) goto FalseExit;
-                bytesWritten = digitCount;
-                buffer = buffer.Slice(0, digitCount);
+                return TryFormatUInt64LessThanQuintillion((ulong)value, buffer, out bytesWritten); 
             }
+        }
 
-            // WriteDigits does not do bounds checks
-            FormattingHelpers.WriteDigits((ulong)value, buffer);
-            return true;
+        // Split long into three parts that can each fit in a uint - {1 digit}{9 digits}{9 digits}
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool TryFormatInt64MoreThanQuintillion(long value, Span<byte> buffer, out int bytesWritten)
+        {
+            if (value < 0)
```

Comment:
Once you remove AggressiveInlining, this check be rather in the place where TryFormatInt64MoreThanQuintillion is called and this method should be only handling negative numbers.